app/page.tsx

```diff
@@ -29,26 +30,52 @@ import {
 } from './filters';
 import { isAddress } from 'viem';
 import Image from 'next/image';
+import { getMarketCategories, getOpenMarkets } from '@/queries/dune';
+import { Categories } from '@/constants';
 
 const ITEMS_PER_PAGE = 12;
 const SEARCH_DEBOUNCE_DELAY = 600;
 const DEFAULT_ORDER_OPTION = orderFilters[0];
 const DEFAULT_STATE_OPTION = stateFilters[0];
 const DEFAULT_CREATOR_OPTION = creatorFilters[0];
 
-enum Categories {
-  TECHNOLOGY = 'technology',
-  CRYPTO = 'crypto',
-  BUSINESS = 'business',
-  POLITICS = 'politics',
-  ECONOMY = 'economy',
-  INTERNATIONAL = 'international',
-  SPORTS = 'sports',
-  ENTERTAINMENT = 'entertainment',
-}
-
 type CategoryOptions = Categories | '';
 
+const groupOpenMarketsByCategory = (data: any) => {
+  if (!data) return [];
+  const marketsGroupedByCategory = Object.groupBy(
+    data,
+    (market: { Category: string }) => market.Category
+  );
+
+  if (!marketsGroupedByCategory) return [];
+
+  const categoryCounts = Object.values(Categories).map(category => {
+    const count = marketsGroupedByCategory[category]?.length || 0;
+    return { [category]: count };
+  });
+
+  return categoryCounts.sort((a, b) => Object.values(b)[0] - Object.values(a)[0]);
+};
+
+const buildSortedMarketArray = (data: any) => {
+  if (!data) return [];
+
+  const allMarketsSortedCategories: { [key: string]: number }[] = [];
+
+  for (const category of Object.values(Categories)) {
+    const market = data.find(
+      ({ category: categoryOption }: { category: string }) => categoryOption === category
+    );
+    const count = market ? market.cnt : 0;
+    allMarketsSortedCategories.push({ [category]: count });
+  }
+
+  allMarketsSortedCategories.sort((a, b) => Object.values(b)[0] - Object.values(a)[0]);
+
+  return allMarketsSortedCategories;
+};
```

Comment:
it would be great if we could improve this 2 functions a bit

- don't think the name are good 
- probably we can simplify a bit of the code, it's not that easy to read.